### PR TITLE
Init

### PR DESCRIPTION
**Pull request by ShiranAvidov**

### CODE REVIEW

File: azure_monitor_metrics_receiver.go [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+
+func NewResource(resourceType string, metrics []string, aggregations []string) *Resource {
+	return &Resource{
+		resourceType: resourceType,
+		metrics: metrics,
+		aggregations: aggregations,
+	}
+}
+
+func newAzureResourcesClient(subscriptionID string, credential *azidentity.ClientSecretCredential) *azureResourcesClient {
```
Comment (yotamloe) [inline azure_monitor_metrics_receiver.go:124]: Why is this function private? Why does this differ from `NewResource` or `NewResourceTarget` in terms of visibility

**Review by yotamloe — CHANGES_REQUESTED** [review]
Overall looks good, I left a few comments about the visibility logic

File: azure_monitor_metrics_receiver.go (hunk 1 above)
Comment (ShiranAvidov) [reply azure_monitor_metrics_receiver.go:124]: I use it for polymorphism by using the interface type I created. basically it is for constructor dependency injection, so I can do mocking for unit tests.

File: README.md [hunk 2]
```diff
@@ -1 +1,93 @@
-# azure-monitor
\ No newline at end of file
+# Azure Monitor Metrics Receiver
+
+An SDK wrapper that uses Azure Monitor SDK. 
+Lets you receive metrics from שAzure resources using Azure Monitor API in 3 ways:
```
[the remark this answers is not in the document]
Comment (ShiranAvidov) [reply README.md]: fixed

File: initializer.go [hunk 3]
```diff
@@ -0,0 +1,581 @@
...
+	return nil
+}
+
+func (ammr *AzureMonitorMetricsReceiver) addPrefixToResourceTargetsResourceID() {
+	for _, target := range ammr.Targets.ResourceTargets {
+		target.ResourceID = "/subscriptions/" + ammr.subscriptionID + "/" + target.ResourceID
+	}
+}
+
+func (ammr *AzureMonitorMetricsReceiver) CreateResourceTargetsFromResourceGroupTargets() error {
```
[the remark this answers is not in the document]
Comment (ShiranAvidov) [reply initializer.go:136]: yeah its part of couple of methods they need to use in init

File: initializer.go [hunk 4]
```diff
@@ -0,0 +1,581 @@
...
+			return fmt.Errorf("error creating resource target from resource group target resources: %v", err)
+		}
+
+		resourceTargetsCreatedNum += currentResourceTargetsCreatedNum
+	}
+
+	return nil
+}
+
+func (ammr *AzureMonitorMetricsReceiver) CreateResourceTargetsFromSubscriptionTargets() error {
```
[the remark this answers is not in the document]
Comment (ShiranAvidov) [reply initializer.go:171]: yeah its part of couple of methods they need to use in init

File: initializer.go [hunk 5]
```diff
@@ -0,0 +1,581 @@
...
+	return nil
+}
+
+func (ammr *AzureMonitorMetricsReceiver) changeResourceTargetsMetricsWithComma() {
+	for _, target := range ammr.Targets.ResourceTargets {
+		target.changeMetricsWithComma()
+	}
+}
+
+func (ammr *AzureMonitorMetricsReceiver) SplitResourceTargetsMetricsByMinTimeGrain() error {
```
[the remark this answers is not in the document]
Comment (ShiranAvidov) [reply initializer.go:272]: yeah its part of couple of methods they need to use in init

**Comment by yotamloe** [comment]
Why this project has an MIT license and not apache2?

**Review by yotamloe — APPROVED** [review]

**Pull request merged**